src/Http/Routing/src/Builder/MinimalActionEndpointRouteBuilderExtensions.cs

```diff
@@ -184,6 +186,25 @@ public static class MinimalActionEndpointRouteBuilderExtensions
             // Add MethodInfo as metadata to assist with OpenAPI generation for the endpoint.
             builder.Metadata.Add(action.Method);
 
+            // Methods defined in a top-level program are generated as statics so the delegate
+            // target will be null. Inline lambdas are compiler generated properties so they can
+            // be filtered that way.
+            if (action.Target == null || !TypeHelper.IsCompilerGenerated(action.Method.Name))
+            {
+                if (GeneratedNameParser.TryParseLocalFunctionName(action.Method.Name, out var endpointName))
+                {
+                    builder.Metadata.Add(new EndpointNameMetadata(endpointName));
+                    builder.Metadata.Add(new RouteNameMetadata(endpointName));
+                    builder.DisplayName = $"{builder.DisplayName} => {endpointName}";
+                }
+                else
+                {
+                    builder.Metadata.Add(new EndpointNameMetadata(action.Method.Name));
+                    builder.Metadata.Add(new RouteNameMetadata(action.Method.Name));
+                    builder.DisplayName = $"{builder.DisplayName} => {endpointName}";
```

Comment:
Should this use `action.Method.Name` this instead?

```suggestion
                    builder.DisplayName = $"{builder.DisplayName} => {action.Method.Name}";
```

If so you could also dedupe this code:

```c#
if (!GeneratedNameParser.TryParseLocalFunctionName(action.Method.Name, out var endpointName))
    endpointName = action.Method.Name;

builder.Metadata.Add(new EndpointNameMetadata(endpointName));
builder.Metadata.Add(new RouteNameMetadata(endpointName));
builder.DisplayName = $"{builder.DisplayName} => {endpointName}";
```